java/code/src/com/redhat/rhn/taskomatic/task/threaded/QueueDriver.java

```diff
@@ -67,4 +67,10 @@
      * Actions that has to be executed, when queue is created
      */
     void initialize();
+
+    /**
+     * Specify if this is a blocking worker thread
+     * @True if the task queue should wait end of the workers before continue
+     */
+    boolean isBlockingTask();
```

Comment:
I like the idea, but maybe we can improve the name :-)

- isBlockingTaskQueue()
- blockTaskQueue()
- waitForEmptyQueue()
- ...